prime-dotnet-webapi/ApiDbContext.cs

```diff
@@ -204,11 +204,27 @@ protected override void OnModelCreating(ModelBuilder modelBuilder)
                 .WithMany(e => e.AdjudicatorNotes)
                 .HasForeignKey(an => an.EnrolleeId);
 
-            modelBuilder.Entity<AccessTerm>()
+            modelBuilder.Entity<Agreement>()
                 .HasOne(toa => toa.Enrollee)
-                .WithMany(e => e.AccessTerms)
+                .WithMany(e => e.Agreements)
                 .HasForeignKey(toa => toa.EnrolleeId);
 
+            modelBuilder.Entity<Agreement>()
+                .HasOne(toa => toa.Organization)
+                .WithMany(e => e.Agreements)
+                .HasForeignKey(toa => toa.OrganizationId);
+
+            modelBuilder.Entity<Agreement>()
+                .HasOne(toa => toa.Party)
+                .WithMany(e => e.Agreements)
+                .HasForeignKey(toa => toa.PartyId);
+
+            modelBuilder.Entity<Agreement>()
+                .HasCheckConstraint("CHK_OnlyOneForeignKey",
```

Comment:
Do Check Constraints have a similar naming convention to PK and FK constraints where the name of the table is in the name? Something like "CHK_Agreement_OnlyOneForeignKey"